### PR TITLE
W-10877267: A.11.2 Implement the automatic propagation of trace context

### PR DESCRIPTION
**Pull request by fsgonz**

### CODE REVIEW

File: modules/extensions-support/src/main/java/org/mule/runtime/module/extension/internal/runtime/parameter/PropagateAllDistributedTraceContextManager.java [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+  }
+
+  @Override
+  public void setRemoteTraceContextMap(Map<String, String> contextMap) {
+    this.contextMap = contextMap;
+  }
+
+  @Override
+  public Map<String, String> getRemoteTraceContextMap() {
+    return contextMap;
```
Comment (marianogonzalez) [inline modules/extensions-support/src/main/java/org/mule/runtime/module/extension/internal/runtime/parameter/PropagateAllDistributedTraceContextManager.java:41]: should use unmodifiableMap() here?

**Review by marianogonzalez — CHANGES_REQUESTED** [review]

File: core/src/main/java/org/mule/runtime/core/internal/event/AbstractEventContext.java [hunk 2]
```diff
@@ -48,7 +49,7 @@
  *
  * @since 4.0
  */
-abstract class AbstractEventContext implements BaseEventContext {
+abstract class AbstractEventContext implements DistributedTraceContextAware, BaseEventContext {
```
[the remark this answers is not in the document]
Comment (fsgonz) [reply core/src/main/java/org/mule/runtime/core/internal/event/AbstractEventContext.java:52]: BaseContext is a privileged interface. I would rather not change anything related to api (even privileged apis), as this is only used internally.

File: core/src/main/java/org/mule/runtime/core/internal/event/DefaultEventContext.java [hunk 3]
```diff
@@ -317,15 +322,23 @@ protected String basicToString() {
         + getOriginatingLocation().getRootContainerName() + " }";
   }
 
+  public DistributedTraceContext getDistributedTraceContext() {
```
Comment (marianogonzalez) [inline core/src/main/java/org/mule/runtime/core/internal/event/DefaultEventContext.java:325]: are these two missing an `@Override`?

File: modules/extensions-support/src/main/java/org/mule/runtime/module/extension/internal/runtime/parameter/PropagateAllDistributedTraceContextManager.java [hunk 4]
```diff
@@ -27,8 +28,7 @@ public PropagateAllDistributedTraceContextManager(DistributedTraceContext distri
   }
 
   private void resolveContextMap(DistributedTraceContext distributedTraceContext) {
-    contextMap = new HashMap<>(distributedTraceContext.tracingFieldsAsMap());
-    contextMap.putAll(distributedTraceContext.baggageItemsAsMap());
+    contextMap = unmodifiableMap(distributedTraceContext.baggageItemsAsMap());
```
Comment (marianogonzalez) [inline modules/extensions-support/src/main/java/org/mule/runtime/module/extension/internal/runtime/parameter/PropagateAllDistributedTraceContextManager.java:31]: is this change correct? It seems like previously you were also including the tracing fields but not anymore. Which version is boggus?

**Review by marianogonzalez — CHANGES_REQUESTED** [review]

**Review by marianogonzalez — APPROVED** [review]

**Pull request merged**